src/arfsdao.ts

```diff
@@ -152,57 +154,115 @@ export class ArFSDAOAnonymous extends ArFSDAOType {
-		tags.forEach((tag: GQLTagInterface) => {
-			const key = tag.name;
-			const { value } = tag;
-			switch (key) {
-				case 'App-Name':
-					driveBuilder.appName = value;
-					break;
-				case 'App-Version':
-					driveBuilder.appVersion = value;
-					break;
-				case 'ArFS':
-					driveBuilder.arFS = value;
-					break;
-				case 'Content-Type':
-					driveBuilder.contentType = value as ContentType;
-					break;
-				case 'Drive-Id':
-					driveBuilder.driveId = value;
-					break;
-				case 'Drive-Privacy':
-					driveBuilder.drivePrivacy = value as DrivePrivacy;
-					break;
-				case 'Entity-Type':
-					driveBuilder.entityType = value as EntityType;
-					break;
-				case 'Unix-Time':
-					driveBuilder.unixTime = +value;
-					break;
-				default:
-					break;
-			}
-		});
+		const driveBuilder = new ArFSPublicDriveBuilder(driveId, this.arweave);
+		return driveBuilder.build();
+	}
+
+	async getPublicFolder(folderId: string): Promise<ArFSPublicFolder> {
+		const gqlQuery = buildQuery([{ name: 'Folder-Id', value: folderId }]);
 
-		// Get the drives transaction ID
-		driveBuilder.txId = node.id;
+		const response = await this.arweave.api.post(graphQLURL, gqlQuery);
 
-		if (driveBuilder.txId) {
-			const txData = await this.arweave.transactions.getData(driveBuilder.txId, { decode: true });
-			const dataString = await Utf8ArrayToStr(txData);
-			const dataJSON = await JSON.parse(dataString);
+		const { data } = response.data;
+		const { transactions } = data;
+		const { edges } = transactions;
 
-			// Get the drive name and root folder id
-			driveBuilder.name = dataJSON.name;
-			driveBuilder.rootFolderId = dataJSON.rootFolderId;
+		if (!edges.length) {
+			throw new Error(`Public folder with Folder ID ${folderId} not found!`);
 		}
 
-		return driveBuilder.build();
+		const folderBuilder = new ArFSPublicFolderBuilder(folderId, this.arweave);
+		return await folderBuilder.build();
+	}
+
+	async getPublicFilesWithParentFolderIds(folderIDs: FolderID[]): Promise<ArFSPublicFile[]> {
+		let cursor = '';
+		let hasNextPage = true;
+		const allFiles: ArFSPublicFile[] = [];
+
+		while (hasNextPage) {
+			const gqlQuery = buildQuery(
+				[
+					{ name: 'Parent-Folder-Id', value: folderIDs },
+					{ name: 'Entity-Type', value: 'file' }
+				],
+				cursor
+			);
+
+			const response = await this.arweave.api.post(graphQLURL, gqlQuery);
+			const { data } = response.data;
+			const { transactions } = data;
+			const { edges } = transactions;
+			hasNextPage = transactions.pageInfo.hasNextPage;
+			const files: Promise<ArFSPublicFile>[] = edges.map(async (edge: GQLEdgeInterface) => {
+				const { node } = edge;
+				cursor = edge.cursor;
+				const fileBuilder = await ArFSPublicFileBuilder.fromArweaveNode(node, this.arweave);
+				return fileBuilder.build(node);
+			});
+			allFiles.push(...(await Promise.all(files)));
+		}
+		return allFiles;
+	}
+
+	async getAllFoldersOfPublicDrive(driveId: DriveID): Promise<ArFSPublicFolder[]> {
```

Comment:
It would be nice to know from the 1) name, 2) a parameter, or 3) a doc block that this function will return all revisions of each folder. For each, that might look like:
1) `getAllRevisionsOfFoldersInPublicDrive`
2) `getAllFoldersOfPublicDrive(driveId: DriveID, latestRevisionsOnly: boolean)`
3) `// Returns all revisions of all folders in the drive with ID driveID`

Reply:
I loved the 2nd one the most: 67408099921350d987712a2badf7af62642c58b9